### PR TITLE
Barkin/es6

### PR DESCRIPTION
**Pull request by volfied**
ES6 ifies spaces-adapter, all sources except `src/lib`.

We have plans on separating src/lib into it's own library in the future.

Important changes:
 - ES6 ification of src folders
 - Webpack to compile the bundle
 - Point to the compiled bundle in package.json `main` field so external importers don't need to resolve sources
 - Get rid of wolfy87-eventEmitter, and use webpack supplied Node.js "events" library instead.

Still a work in progress (but can be merged for use), remaining tasks:
- [x] Fix the unit tests
- [ ] ES6ify lib folder, preferably in it's own project and take it out of spaces-adapter

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -48,7 +48,6 @@
     "disallowSpacesInsideParentheses": true,
     "disallowTabs": true,
     "disallowTrailingComma": true,
-    "disallowTrailingWhitespaceInSource": true,
```
Comment (iwehrman) [inline .jscsrc]: Why?

File: package.json [hunk 2]
```diff
@@ -27,9 +27,18 @@
   "bugs": {
     "url": "https://github.com/adobe-photoshop/spaces-adapater/issues"
   },
-  "main": "src/main.js",
+  "main": "build/spaces-adapter.js",
   "homepage": "https://github.com/adobe-photoshop/spaces-adapater",
```
Comment (iwehrman) [inline package.json]: Not your bug, but: typo.

File: package.json [hunk 3]
```diff
@@ -27,9 +27,18 @@
   "bugs": {
     "url": "https://github.com/adobe-photoshop/spaces-adapater/issues"
```
Comment (iwehrman) [inline package.json]: And here.

**Comment by volfied** [comment]
@iwehrman Thanks for review, I must have copied the jscs file from my other repo and missed that small change.

**Comment by volfied** [comment]
This PR now also includes ES6ified OWL and Window APIs

**Comment by shaoshing** [comment]
@volfied I re-compile a unminified build for easier debugging in dev, also to better support mocking the internal functions as they will be renamed if minified version.

**Comment by mcilroyc** [comment]
alright, v2 it is!

**Pull request merged**